### PR TITLE
feat: add unix domain socket file support

### PR DESCRIPTION
**Pull request by XadillaX**

### CODE REVIEW

**Review by atian25 — APPROVED** [review]

**Review by killagu — APPROVED** [review]
LGTM

**Review by dead-horse — APPROVED** [review]

File: lib/urllib.js [hunk 1]
```diff
@@ -315,6 +316,9 @@ function requestWithCallback(url, args, callback) {
       options.headers[key] = args.headers[name];
     }
   }
+  if (args.socketPath) {
+    options.socketPath = args.socketPath;
```
Comment (dead-horse) [inline lib/urllib.js:320]: README 里面补充一下吧

Comment (XadillaX) [reply lib/urllib.js:320]: done

**Review by dead-horse — APPROVED** [review]

**Comment by dead-horse** [comment]
2.37.0

**Pull request merged**